### PR TITLE
[v3.6] Don't rename webview& video c++ destuctor to destroy in webview/video.ini, add a 'destroy' method and bind it instead.

### PR DESCRIPTION
**Pull request by dumganhar**
### Changelog

* 

-------

### Continuous Integration

This pull request:

* [X] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/ui/videoplayer/VideoPlayer.h [hunk 1]
```diff
@@ -66,6 +66,12 @@ class VideoPlayer : public RefCounted {
     };
 
     VideoPlayer();
+    ~VideoPlayer();
```
Comment (minggo) [inline native/cocos/ui/videoplayer/VideoPlayer.h:69]: override needed?

**Review by PatriceJiang — APPROVED** [review]

File: native/cocos/ui/webview/WebView-inl.h [hunk 2]
```diff
@@ -58,75 +58,118 @@ WebView *WebView::create() {
     return nullptr;
 }
 
+void WebView::destroy() {
+    if (_impl != nullptr) {
+        _impl->destroy();
+        _impl = nullptr;
+    }
```
Comment (minggo) [inline native/cocos/ui/webview/WebView-inl.h]: Just use `CC_SAFE_DESTROY()` ？

File: native/cocos/ui/videoplayer/VideoPlayer.h (hunk 1 above)
Comment (dumganhar) [reply native/cocos/ui/videoplayer/VideoPlayer.h:69]: No, It should not be inherited from `RefCounted`. I will remove that.

File: native/cocos/ui/webview/WebView-inl.h (hunk 2 above)
Comment (dumganhar) [reply native/cocos/ui/webview/WebView-inl.h]: Seems better. :)

File: native/cocos/ui/videoplayer/VideoPlayer-ios.mm [hunk 3]
```diff
@@ -298,8 +298,13 @@ - (void)addPlayerControllerSubView {
...
 
 VideoPlayer::~VideoPlayer() {
-    if (_videoView) {
-        [((UIVideoViewWrapperIos *)_videoView) dealloc];
+    destroy();
+}
+
+void VideoPlayer::destroy() {
+    if (_videoView != nil) {
+        [((UIVideoViewWrapperIos *)_videoView) release];
```
Comment (dumganhar) [inline native/cocos/ui/videoplayer/VideoPlayer-ios.mm:306]: Don't invoke dealloc directly.

**Review by minggo — APPROVED** [review]

**Pull request merged**